parsers/SG.py

```diff
@@ -58,6 +61,41 @@
 }
 
 
+def get_solar(session=None):
+    """
+    Fetches a graphic showing estimated solar production data.
+    Uses OCR (tesseract) to extract MW value.
+    Returns a float or None.
+    """
+
+    s = session or requests.Session()
+    url = 'https://www.ema.gov.sg/cmsmedia/irradiance/plot.png'
+    solar_image = Image.open(s.get(url, stream=True).raw)
+
+    gray = solar_image.convert('L')
+    threshold_filter = lambda x: 0 if x<77 else 255
+    black_white = gray.point(threshold_filter, '1')
+
+    text = image_to_string(black_white, lang='eng')
+
+    pattern = r'Est. PV Output: (.*)MWac'
+    val = re.search(pattern, text, re.MULTILINE).group(1)
+
+    # At night format changes from 0.00 to 0
+    # tesseract cannot distinguish singular 0 and O in font provided by image.
+    # This try/except will make sure no invalid data is returned.
+    try:
+        solar = float(val)
+    except ValueError as err:
+        if len(val) == 1 and 'O' in val:
+            solar = 0.0
+        else:
+            print("Singapore solar data is unreadable - got {}.".format(val))
```

Comment:
maybe raise an exception is in case?

Reply:
If we raise exception, we lose data from main parser - there is no history backfill in SG

Reply:
got it